External/FEXCore/Source/Interface/IR/Passes/ConstProp.cpp

```diff
@@ -893,6 +904,33 @@ bool ConstProp::ConstantPropagation(IREmitter *IREmit, const IRListView& Current
         IREmit->ReplaceWithConstant(CodeNode, NewConstant);
         Changed = true;
       }
+      else if (SrcIsConstant) {
+        // If the source is constant in a BFI then it can be converted to an or or bic depending.
+        if (ConstantSrc == 0) {
+          // We are wanting to clear the bitfield.
+          IREmit->SetWriteCursor(CodeNode);
+          uint64_t SourceMask = (1ULL << Op->Width) - 1;
+          if (Op->Width == 64)
+            SourceMask = ~0ULL;
+
+          uint64_t NewConstant = SourceMask << Op->lsb;
+
+          auto andn = IREmit->_Andn(CurrentIR.GetNode(Op->Header.Args[0]), IREmit->_Constant(NewConstant));
+          andn.first->Header.Size = IROp->Size;
+          IREmit->ReplaceAllUsesWith(CodeNode, andn);
+          Changed = true;
+        }
+        else {
+          // We are trying to insert constant, if it is a bitfield of only set bits then we can orr it.
+          if (HasConsecutiveBits(ConstantSrc, Op->Width)) {
+            IREmit->SetWriteCursor(CodeNode);
+            auto orr = IREmit->_Or(CurrentIR.GetNode(Op->Header.Args[0]), IREmit->_Constant(ConstantSrc << Op->lsb));
```

Comment:
I think `ConstantSrc << Op->lsb` would also have the wrong value if there are any non-zero bits in `ConstantSrc` past `Op->Width`

Reply:
Resolved this as mentioned in the other comment. Generating the same constant on both halves of andn/orr paths since it actually uses the same constant. Which removes this weirdness.